inginious/frontend/pages/utils.py

```diff
@@ -46,7 +46,11 @@ def app(self):
         """ Returns the web application singleton """
         return flask.current_app
 
-    def _pre_check(self, sessionid):
+    def dispatch_request(self, **kwargs):
+        kwargs['session_id'] = flask.request.args.get('session_id')
+        return super().dispatch_request(**kwargs)
```

Comment:
This is the way I hooked before a page is being called to extract the session_id in the same way that the Converter was doing on the path